sihl/src/database_migration_repo.ml

```diff
@@ -23,79 +23,91 @@ end
 module type Sig = sig
   module Migration = Migration
 
-  val create_table_if_not_exists : unit -> unit Lwt.t
-  val get : namespace:string -> Migration.t option Lwt.t
-  val get_all : unit -> Migration.t list Lwt.t
-  val upsert : state:Migration.t -> unit Lwt.t
+  val create_table_if_not_exists : string -> unit Lwt.t
+  val get : string -> namespace:string -> Migration.t option Lwt.t
+  val get_all : string -> Migration.t list Lwt.t
+  val upsert : string -> Migration.t -> unit Lwt.t
 end
 
+(* Common functions *)
+let get_request table =
+  Caqti_request.find_opt
+    Caqti_type.string
+    Caqti_type.(tup3 string int bool)
+    (Format.sprintf
+       {sql|
+       SELECT
+         namespace,
+         version,
+         dirty
+       FROM %s
+       WHERE namespace = ?;
+       |sql}
+       table)
+;;
+
+let get table ~namespace =
+  Database.query (fun (module Connection : Caqti_lwt.CONNECTION) ->
+      Connection.find_opt (get_request table) namespace
+      |> Lwt.map Database.raise_error)
+  |> Lwt.map (Option.map Migration.of_tuple)
+;;
+
+let get_all_request table =
+  Caqti_request.collect
+    Caqti_type.unit
+    Caqti_type.(tup3 string int bool)
+    (Format.sprintf
+       {sql|
+       SELECT
+         namespace,
+         version,
+         dirty
+       FROM %s;
```

Comment:
Could this be an attack vector for SQL injection?

Reply:
No, the table name is not based on any end user input. You need access to the env variables to set it.

Reply:
So 
```ocaml
  val create_table_if_not_exists : string -> unit Lwt.t
  val get : string -> namespace:string -> Migration.t option Lwt.t
  val get_all : string -> Migration.t list Lwt.t
  val upsert : string -> Migration.t -> unit Lwt.t
  ```
are not exposed to sihl user?

Reply:
No this is just the repository, the service API is what matters.

Reply:
Even if it was exposed to a sihl developer, I think we could just mention it in the documentation of the signature to not further expose it to a end user.

Reply:
That's fair, thanks